modules/core/shared/src/main/scala/lucuma/core/math/Epoch.scala

```diff
@@ -37,13 +44,13 @@ final class Epoch private (val scheme: Epoch.Scheme, val toMilliyears: Int) {
 
   /** Offset in epoch-years from this `Epoch` to the given fractional Julian day. */
   def untilJulianDay(jd: Double): Double =
-    untilEpochYear(scheme.fromJulianDay(jd).epochYear)
+    untilEpochYear(scheme.fromJulianDayToEpochYears(jd))
 
   /** Offset in epoch-years from this `Epoch` to the given epoch year under the same scheme. */
   def untilEpochYear(epochYear: Double): Double =
     epochYear - this.epochYear
 
-  def plusYears(y: Double): Epoch =
+  def plusYears(y: Double): Option[Epoch] =
```

Comment:
Would it make sense to make this always take a type positive such that it returns always a valid `Epoch`?

Reply:
Seems reasonable to me. Not urgent but maybe add a story for this.